### PR TITLE
[Java/Spring] all-of and one-of Improvements and Fixes

### PR DESCRIPTION
**Pull request by cachescrubber**
# [Java/Spring] all-of and one-of Improvements and Fixes

This PR fixes a Bug in `OneOfImplementorAdditionalData` which pulls in parent properties in a one-of implementor when
the one-of implementor also inherits form a parent using all-of.

In order to use the fluent setter methods for models using inheritance, all Models are post processed and the
`CodegenModel.parentVars` are set up to recursive include all parent properties incl. all necessary imports.

### PR checklist
 
- [ x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [ x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [ x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [ x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by cachescrubber** [comment]
FYI @wing328  (2022/02) @welshm (2022/02) @MelleD (2022/02) @atextor (2022/02) @manedev79 (2022/02) @javisst (2022/02) @borsch (2022/02) @banlevente (2022/02)

File: samples/openapi3/server/petstore/spring-boot-oneof/src/main/java/org/openapitools/model/BarRef.java [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+    return true;
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(href, atSchemaLocation, atBaseType, atTypesuper.hashCode());
```
Comment (cachescrubber) [inline samples/openapi3/server/petstore/spring-boot-oneof/src/main/java/org/openapitools/model/BarRef.java]: The Bug caused by `org.openapitools.codegen.utils.OneOfImplementorAdditionalData#addToImplementor`

**Review by borsch — APPROVED** [review]

File: samples/openapi3/server/petstore/spring-boot-oneof/src/main/java/org/openapitools/model/BarRef.java [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+  private String href;
+
+  @JsonProperty("@schemaLocation")
+  private String atSchemaLocation;
+
+  @JsonProperty("@baseType")
+  private String atBaseType;
+
+  @JsonProperty("@type")
+  private String atType;
```
Comment (cachescrubber) [inline samples/openapi3/server/petstore/spring-boot-oneof/src/main/java/org/openapitools/model/BarRef.java]: This properties are pulled in from EntityRef (and Entity) (caused by org.openapitools.codegen.utils.OneOfImplementorAdditionalData#addFromInterfaceModel).

File: samples/openapi3/server/petstore/spring-boot-oneof/src/main/java/org/openapitools/model/PizzaSpeziale.java [hunk 3]
```diff
@@ -8,6 +8,7 @@
 import com.fasterxml.jackson.annotation.JsonSubTypes;
 import com.fasterxml.jackson.annotation.JsonTypeInfo;
 import org.openapitools.model.Pizza;
+import java.math.BigDecimal;
```
Comment (cachescrubber) [inline samples/openapi3/server/petstore/spring-boot-oneof/src/main/java/org/openapitools/model/PizzaSpeziale.java:11]: An import needed for the fluent setter.

File: samples/openapi3/server/petstore/spring-boot-oneof/src/main/java/org/openapitools/model/BarRef.java [hunk 4]
```diff
@@ -26,6 +26,41 @@
...
+    return this;
+  }
+
+  public BarRef atSchemaLocation(String atSchemaLocation) {
+    super.setAtSchemaLocation(atSchemaLocation);
+    return this;
+  }
+
+  public BarRef atBaseType(String atBaseType) {
+    super.setAtBaseType(atBaseType);
```
Comment (cachescrubber) [inline samples/openapi3/server/petstore/spring-boot-oneof/src/main/java/org/openapitools/model/BarRef.java:55]: In order to not break the fluent setter chain, the inherited properties are exposed as a method in the child class.

**Comment by cachescrubber** [comment]
Replaces by #12089 after merge problems.

**Pull request closed without merging**